### PR TITLE
Dashrews/ROX-13082 UUID protos generated PR 4 of 4

### PR DESCRIPTION
**Pull request by dashrews78**
## Description
This is PR 4 of a 4 PR set.
PR 1:  https://github.com/stackrox/stackrox/pull/3679
PR 2: https://github.com/stackrox/stackrox/pull/3681
PR 3: https://github.com/stackrox/stackrox/pull/3694

The previous PRs dealt with code generation, searcher changes, test constants and fixtures.  
This PR deals with the utilization of all that other work.  

- Protos were updated to use `type(uuid)` for fields that are UUIDs. 
- The stores impacted by those protos were re-generated to deal with the UUID type.  
- Search code was updated and adjusted to handle postgres UUIDs.
- Tests were added that use test proto and such within the searcher code to drive the test cases.
- Additionally per a comment from @md2119 on PR #3679, I have added the support for regex matching since we can regex match on these columns today and it was essentially as simple as using ::text on the field.

There are a lot of files here, but that vast majority of them are generated.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Added some UT to cover various UUID paths.

### CODE REVIEW

**Review by janisz — APPROVED** [review]

**Review by c-du — COMMENTED** [review]
Overall looks good to me.

File: proto/storage/process_indicator.proto [hunk 1]
```diff
@@ -11,24 +11,24 @@ package storage;
 
 // Next available tag: 13
 message ProcessIndicator {
-    // A unique uuid for the Indicator message
-    string id                                      = 1 [(gogoproto.moretags) = 'search:"Process ID,store,hidden"  sql:"pk"'];
+    // A unique type(uuid) for the Indicator message
```
[the remark this answers is not in the document]
Comment (dashrews78) [reply proto/storage/process_indicator.proto:14]: ah yes, replace all got me on one of the iterations.

File: proto/storage/process_indicator.proto [hunk 2]
```diff
@@ -11,24 +11,24 @@ package storage;
...
+    string id                                      = 1 [(gogoproto.moretags) = 'search:"Process ID,store,hidden"  sql:"pk,type(uuid)"'];
 
-    string deployment_id                           = 2 [(gogoproto.moretags) = 'search:"Deployment ID,store,hidden" policy:",prefer-parent" sql:"index=hash,fk(Deployment:id),no-fk-constraint"'];
+    string deployment_id                           = 2 [(gogoproto.moretags) = 'search:"Deployment ID,store,hidden" policy:",prefer-parent" sql:"index=hash,fk(Deployment:id),no-fk-constraint,type(uuid)"'];
     reserved 10; // Previously deployment_state_ts
 
     string container_name                          = 3 [(gogoproto.moretags) = 'search:"Container Name,hidden" policy:",prefer-parent"'];
     // Pod name
     string pod_id                                  = 4 [(gogoproto.moretags) = 'search:"Pod ID,hidden"'];
```
[the remark this answers is not in the document]
Comment (dashrews78) [reply proto/storage/process_indicator.proto:22]: @rhybrillou I had checked that out before.  I'm not entirely sure why there are both fields, but pod_id is not a UUID and pod_uid is.  In the code pod_uid is the one used to reference the id in the pods table. 
`central_active=# select id, podid, poduid from process_indicators;
                  id                  |                      podid                       |                poduid                
--------------------------------------+--------------------------------------------------+--------------------------------------
 64a96d48-a800-5c7d-9379-1fe2668e2a29 | d6d1f550603c32897ae8b43c341d780ed816b6196fa353fa | 0185ac3d-6fff-5bfc-890c-a84a406fd652`

**Review by md2119 — COMMENTED** [review]
Seems like some changes in this are already captured in part 3/4. Would make the reviewing a lot easier if you could rebase.

**Review by c-du — APPROVED** [review]

File: proto/storage/process_indicator.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (dashrews78) [reply proto/storage/process_indicator.proto:22]: @md2119 yes, it appears the comment is correct.

```
central_active=# select p.id, p.name, pi.podid, pi.poduid from pods p inner join process_indicators pi on p.id = pi.poduid;
                  id                  |                       name                       |                      podid                       |                poduid                
--------------------------------------+--------------------------------------------------+--------------------------------------------------+--------------------------------------
 234e886d-86d9-58bd-bfda-2c2f7ea31725 | c8e42380aff61c233a68766a8d8047f92e5691f775c12117 | c8e42380aff61c233a68766a8d8047f92e5691f775c12117 | 234e886d-86d9-58bd-bfda-2c2f7ea31725
 ```

**Pull request merged**